packages/components/src/styles/components/code-block/index.scss

```diff
@@ -125,13 +126,63 @@ $hds-code-block-code-padding: 16px;
   border-radius: inherit;
 }
 
+.hds-code-block__overlay-footer {
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 10px $hds-code-block-code-padding 10px $hds-code-block-code-padding;
+  pointer-events: none; // fix issue with scrolling when hovering over the footer
+
+  // re-enable pointer events for the button (or any content inside the footer)
+  > * {
+    pointer-events: auto;
+  }
+}
+
+.hds-code-block--has-overlay-footer {
+  .hds-code-block__code::after {
+    // gradient element
+    position: sticky; // prevent gradient from scrolling together with content
+    bottom: -$hds-code-block-code-padding;
+    display: block;
+    width: calc(100% + #{$hds-code-block-code-padding * 2});
+    height: $hds-code-block-code-footer-height;
+    background: linear-gradient(360deg, #0d0e12 37.07%, rgba(13, 14, 18, 25%) 100%);
+    transform: translateX(-$hds-code-block-code-padding);
+    content: "";
+    pointer-events: none;
+  }
+}
+
+.hds-code-block--has-overlay-footer-expanded {
```

Comment:
The naming of this class name seems confusing. To me, it sounds like the footer itself is expanded somehow.

How about changing it to just ".hds-code-block--is-expanded"?